src/client/datascience/jupyter/kernels/types.ts

```diff
@@ -147,7 +147,7 @@ export interface IKernel extends IAsyncDisposable {
     interrupt(): Promise<InterruptResult>;
     restart(): Promise<void>;
     executeCell(cell: NotebookCell): Promise<NotebookCellRunState>;
-    executeHidden(code: string): Promise<nbformat.IOutput[]>;
+    executeHidden(code: string, doc: NotebookDocument): Promise<nbformat.IOutput[]>;
```

Comment:
We don't need to pass the document

Reply:
It's already available in the class

Reply:
We need it [here](https://github.com/microsoft/vscode-jupyter/blob/9eaa2cecebf0e9616af795527488b524f089606b/src/client/datascience/jupyter/kernels/kernelExecution.ts#L87) to get the executionQueue

Reply:
I don't see it in the class, there's `documentExecutions` which uses documents as keys, but there's no document in the class

Reply:
Check line 113 of kernel.ts file